iox_query_influxql/src/plan/planner.rs

```diff
@@ -816,29 +890,121 @@ impl<'a> InfluxQLToLogicalPlan<'a> {
         // Wrap the plan in a `LogicalPlan::Projection` from the select expressions
         let plan = project(plan, select_exprs)?;
 
-        if matches!(
-            ctx.projection_type,
-            ProjectionType::WindowAggregate | ProjectionType::Window
-        ) {
-            // InfluxQL OG physical operators for
+        // InfluxQL OG physical operators for
 
-            // generate a predicate to filter rows where all field values of the row are `NULL`,
-            // like:
-            //
-            //   NOT (field1 IS NULL AND field2 IS NULL AND ...)
-            match conjunction(fields.iter().filter_map(|f| {
-                if matches!(f.data_type, Some(InfluxColumnType::Field(_))) {
-                    Some(f.name.as_expr().is_null())
-                } else {
-                    None
-                }
-            })) {
-                Some(expr) => LogicalPlanBuilder::from(plan).filter(expr.not())?.build(),
-                None => Ok(plan),
+        // generate a predicate to filter rows where all field values of the row are `NULL`,
+        // like:
+        //
+        //   NOT (field1 IS NULL AND field2 IS NULL AND ...)
+        match conjunction(fields.iter().filter_map(|f| {
+            if matches!(f.data_type, Some(InfluxColumnType::Field(_))) {
+                Some(f.name.as_expr().is_null())
+            } else {
+                None
             }
+        })) {
+            Some(expr) => LogicalPlanBuilder::from(plan).filter(expr.not())?.build(),
+            None => Ok(plan),
+        }
+    }
+
+    fn project_select_top_bottom_selector(
+        &self,
+        ctx: &Context<'_>,
+        input: LogicalPlan,
+        fields: &[Field],
+        group_by_tag_set: &[&str],
+    ) -> Result<LogicalPlan> {
+        let schemas = Schemas::new(input.schema())?;
+
+        let (selector_index, call) = fields
+            .iter()
+            .enumerate()
+            .find_map(|f| match &f.1.expr {
+                IQLExpr::Call(c) if c.name == "top" || c.name == "bottom" => Some((f.0, c.clone())),
+                _ => None,
+            })
```

Comment:
```suggestion
            .find_map(|(idx, f)| match &f.expr {
                IQLExpr::Call(c) if c.name == "top" || c.name == "bottom" => Some((idx, c.clone())),
                _ => None,
            })
```

It is idiomatic to deconstruct the tuple into separate variables.